src/java/org/apache/cassandra/db/compaction/CompactionController.java

```diff
@@ -79,6 +79,11 @@ public CompactionController(ColumnFamilyStore cfs, Set<SSTableReader> compacting
              cfs.getCompactionStrategyManager().getCompactionParams().tombstoneOption());
     }
 
+    /**
+     * Make sure all parameters are properly initialized
+     *
+     * Do not call ignoreOverlaps() in constructor
```

Comment:
I think something like:
```
When making changes to the method, be aware that some of the state of the controller may still be uninitialized
(e.g. TWCS sets up the value of ignoreOverlaps() after this completes).
```
in the code block (rather than the JavaDoc) will be better.